src/envelope.rs

```diff
@@ -11,31 +12,82 @@ pub(crate) struct Envelope {
 
 #[derive(Debug)]
 pub(crate) enum Protocol {
+    Tcp(Segment),
     Udp(Datagram),
 }
 
 #[derive(Debug)]
 pub(crate) struct Datagram(pub Bytes);
 
+/// This is a parody of TCP.
+///
+/// We implement just enough to make it realistic, but we skip a ton of
```

Comment:
Realistic is probably the wrong term (& I know you copied this comment from some nonsense I wrote).

I think the goal is fidelity and the ability to simulate real world scenarios. So, checksums are not something we implement because in the real world TCP does checksums for us, and so there is nothing for us to test because we trust TCP will do its job.

Reply:
I took another pass on this.